server/s3_post_complete_action.cc

```diff
@@ -937,13 +933,15 @@ void S3PostCompleteAction::startcleanup() {
       // mark old OID for deletion in overwrite case, this optimizes
       // backgrounddelete decisions.
       ACTION_TASK_ADD(S3PostCompleteAction::mark_old_oid_for_deletion, this);
-      ACTION_TASK_ADD(S3PostCompleteAction::delete_old_object, this);
+      // ACTION_TASK_ADD(S3PostCompleteAction::delete_old_object, this);
     }
+    ACTION_TASK_ADD(S3PostCompleteAction::remove_new_oid_probable_record, this);
   } else if (s3_post_complete_action_state ==
              S3PostCompleteActionState::abortedSinceValidationFailed) {
     // Abort is due to validation failures in part sizes 1..n
     ACTION_TASK_ADD(S3PostCompleteAction::mark_new_oid_for_deletion, this);
-    ACTION_TASK_ADD(S3PostCompleteAction::delete_new_object, this);
+    ACTION_TASK_ADD(S3PostCompleteAction::remove_old_oid_probable_record, this);
+    // ACTION_TASK_ADD(S3PostCompleteAction::delete_new_object, this);
```

Comment:
Delete if it is concurrent delete

Reply:
made some changes @knrajnambiar76 @nileshgovande please take a look